use-cases/school-timetabling/src/main/resources/META-INF/resources/app.js

```diff
@@ -201,73 +283,84 @@ function refreshSolvingButtons(solving) {
 }
 
 function stopSolving() {
-  $.post("/timeTable/stopSolving", function () {
+  $.delete("/timetables/" + scheduleId, function () {
     refreshSolvingButtons(false);
-    refreshTimeTable();
+    refreshSchedule();
   }).fail(function (xhr, ajaxOptions, thrownError) {
     showError("Stop solving failed.", xhr);
   });
 }
 
-function addLesson() {
-  var subject = $("#lesson_subject").val().trim();
-  $.post("/lessons", JSON.stringify({
-    "subject": subject,
-    "teacher": $("#lesson_teacher").val().trim(),
-    "studentGroup": $("#lesson_studentGroup").val().trim()
-  }), function () {
-    refreshTimeTable();
-  }).fail(function (xhr, ajaxOptions, thrownError) {
-    showError("Adding lesson (" + subject + ") failed.", xhr);
-  });
-  $('#lessonDialog').modal('toggle');
-}
-
-function deleteLesson(lesson) {
-  $.delete("/lessons/" + lesson.id, function () {
-    refreshTimeTable();
-  }).fail(function (xhr, ajaxOptions, thrownError) {
-    showError("Deleting lesson (" + lesson.name + ") failed.", xhr);
-  });
+function convertToId(str) {
+  // Base64 encoding without padding to avoid XSS
+  return btoa(str).replace(/=/g, "");
 }
 
-function addTimeslot() {
-  $.post("/timeslots", JSON.stringify({
-    "dayOfWeek": $("#timeslot_dayOfWeek").val().trim().toUpperCase(),
-    "startTime": $("#timeslot_startTime").val().trim(),
-    "endTime": $("#timeslot_endTime").val().trim()
-  }), function () {
-    refreshTimeTable();
-  }).fail(function (xhr, ajaxOptions, thrownError) {
-    showError("Adding timeslot failed.", xhr);
-  });
-  $('#timeslotDialog').modal('toggle');
-}
+function textToClipboard(id) {
+  var text = $("#" + id).text().trim();
 
-function deleteTimeslot(timeslot) {
-  $.delete("/timeslots/" + timeslot.id, function () {
-    refreshTimeTable();
-  }).fail(function (xhr, ajaxOptions, thrownError) {
-    showError("Deleting timeslot (" + timeslot.name + ") failed.", xhr);
-  });
+  var dummy = document.createElement("textarea");
+  document.body.appendChild(dummy);
+  dummy.value = text;
+  dummy.select();
+  document.execCommand("copy");
+  document.body.removeChild(dummy);
 }
 
-function addRoom() {
-  var name = $("#room_name").val().trim();
-  $.post("/rooms", JSON.stringify({
-    "name": name
-  }), function () {
-    refreshTimeTable();
-  }).fail(function (xhr, ajaxOptions, thrownError) {
-    showError("Adding room (" + name + ") failed.", xhr);
-  });
-  $("#roomDialog").modal('toggle');
-}
+// TODO: move to the webjar
+function replaceQuickstartTimefoldAutoHeaderFooter() {
```

Comment:
The copy of the header replacer is intentional to avoid the dependency on the webjar in early phase of this work.